src/main/java/uk/gov/hmcts/reform/sscs/service/hmc/topic/HearingUpdateService.java

```diff
@@ -97,4 +102,30 @@ public void setHearingStatus(String hearingId, @Valid SscsCaseData sscsCaseData,
 
         hearing.getValue().setHearingStatus(hearingStatus);
     }
+
+    public void setWorkBasketFields(String hearingId, @Valid SscsCaseData sscsCaseData, HmcStatus hmcStatus) {
+
+        WorkBasketFields workBasketFields = sscsCaseData.getWorkBasketFields();
+
+        if (isCaseListed(hmcStatus)) {
+            String hearingEpimsId = getHearingEpimsId(hearingId, sscsCaseData);
+            workBasketFields.setHearingEpimsId(hearingEpimsId);
+        } else {
+            workBasketFields.setHearingEpimsId(null);
```

Comment:
Maybe an early return on this if(!isCaseListed(hmcStatus){ return; }

Reply:
It needs to be set to null in that case, not return.

Reply:
Sorry  if(!isCaseListed(hmcStatus){ workBasketFields.setHearingEpimsId(null); return; } <- I meant this

Reply:
I'd perfer not to do that in this case, it doesn't give any benefits and I think this is clearer.